### PR TITLE
test: Upgrade `vitest` (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

**Review by netroy — CHANGES_REQUESTED** [review]
can you also update this in `editor-ui` please?

File: pnpm-lock.yaml
Comment (netroy) [inline pnpm-lock.yaml:3423]: I'm not sure if this would cause any issues. but to be on the safer side, perhaps we should fix the versions of `typescript` and `@types/node` in `pnpm.overrides` in the root package.json

Comment (netroy) [reply pnpm-lock.yaml:3423]: I gave it a try. https://github.com/n8n-io/n8n/commit/cb64aec507cc56ad145917a3c41d91046d0befdb

Comment (ivov) [reply pnpm-lock.yaml:3423]: Checked and tests keep working well in your branch. Only comment is that in your branch you're pinning TS to 4.9.4 but right now we're on 4.8.4. We cannot keep TS as it is?

Comment (netroy) [reply pnpm-lock.yaml:3423]: I updated typescript as we are already pulling in `4.9.4` via transitive dependencies, and when testing I didn't find any issues with the upgrade.
If we were to fix the typescript version to `4.8.4`, all those dependencies will also get `4.8.4`. Did a quick test, and that works as well.

Comment (netroy) [reply pnpm-lock.yaml:3423]: Just FYI: I tested with `4.9.4` even more, and did not see any new issues.

Comment (netroy) [reply pnpm-lock.yaml:3423]: If you want, I can create a new PR with that change, and we can merge this PR then as it is.

Comment (ivov) [reply pnpm-lock.yaml:3423]: Sorry for the delay! Sounds great, let's keep this on hold until your PR is merged. Looking forward to using `satisfies`.

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.215.1`

**Pull request merged**
